symplyphysics/laws/relativistic/lorentz_transformation_of_time.py

```diff
@@ -0,0 +1,72 @@
+from sympy import Eq, sqrt
+from sympy.physics.units import speed_of_light
+from symplyphysics import (
+    units,
+    Quantity,
+    Symbol,
+    validate_input,
+    validate_output,
+)
+from symplyphysics.core.expr_comparisons import expr_equals
+
+# Description
+## Let us consider two inertial reference frames `S` (lab frame) and `S'` (proper frame). The proper frame moves
+## with speed `v` relative to the lab frame. Both frames contain identical fixed (relative to their own frame)
+## clocks that are synchronized according to the [Einstein rule](https://en.wikipedia.org/wiki/Einstein_synchronisation).
+## Let `x, y, z, t` be the coordinates and time of some event in frame `S`, and `x', y', z', t'` be the coordinates
+## and time of the same event in frame `S'`. Assuming that the space is uniform and isotropic and that the time is
+## uniform, there exists a linear dependence between `x, y, z, t` and `x', y', z', t'`, which is called the Lorentz
+## transformation of space and time.
+
+# Law: t' = (t - (V * x) / c**2) / sqrt(1 - (V / c)**2)
```

Comment:
It's a bit weird that time dilation depends on position in stationary frame.

Reply:
Well, it's just how physics works. This is ultimately the manifestation of the fact that space and time are interconnected. And there's a formal derivation of the Lorentz transformation, if it still bugs you.

Reply:
It's not like I doubt this law is correct.
The reason for my confusion is that this law describes a conversion of 4-dimensional point from one frame to another. And here law looks like it describes time flow in another reference frame.
I suggest to somehow explain that this law is about one component of a point. Maybe even put all 4 components in one law.
There is also an idea of making it not a law, but coordinate transformation function in the `core`.

Reply:
Yes, these are in fact 4-vectors and they require separate machinery. Would be a good thing to implement them later on.